bats/scripts/rename-logs.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+set -o errexit -o nounset -o pipefail
+
+# GitHub restricts artifact filenames:
+
+# Invalid characters include: Double quote ", Colon :, Less than <,
+# Greater than >, Vertical bar |, Asterisk *, Question mark ?, Carriage
+# return \r, Line feed \n
+#
+# The following characters are not allowed in files that are uploaded
+# due to limitations with certain file systems such as NTFS. To maintain
+# file system agnostic behavior, these characters are intentionally not
+# allowed to prevent potential problems with downloads on different file
+# systems.
+
+if [[ -z ${1:-} ]]; then
+    echo "usage: $0 LOGDIR"
+    exit 1
+fi
+
+# Find all files and put the names into the FILES array
+# This convoluted form is necessary for bash-3 compatibility.
+# No idea why the command files with errexit enabled.
+set +o errexit
+IFS=$'\n' read -d '' -r -a FILES <<EOF
+$(find "$1" -type f)
```

Comment:
I believe this is typically expressed as
```bash
while read -d $'\0' -r FILE ; do
   NEW=…
done < <(find "$1" -type f -print0)
```
That, at least, seems to work with errexit set?
(On Linux, at least, the only bytes not valid in a file name are `/` and null…)

Reply:
Ah; in that case, please note that we're doing it this way for that reason?

Reply:
Of course, we can also construct the array in the `while read…` loop and then act on it in one go later instead.